### DISCUSSION TITLE
Nomadnet private group messages

### DISCUSSION THREAD

**Discussion by faragher** [markqvist/Reticulum#287] [Nomad Network]
Please note, I said private, not secure. I'll get to that.

I had an idea while going through the NN message board code (which is amazing by the way). By combining the .allowed feature with the message board you could create an ad-hoc private message board to allow group chats. However, this is insecure in so far as there exists a plain-text version of the page on the server (which is typical of old BBS and phpBB implementations, but largely unacceptable today).

Is there any real call for this sort of private message board? It is, at the very least, very simple to implement.

The other concept is a server side repeater. Get a core page with a .allowed that would allow configuring a user name and options, but have that linked to a server script that would take a message sent to RepeaterIdent and send it back to everyone except the original recipient, tagged with <UserFriendlyname>: <OriginalMessage>. It would be processed in the clear on the server, but would otherwise be protected by the standard encryption methods. This provides the benefits of not only a single identity to broadcast to, but also full compatibility with LXMF. It may actually be secure if you count the time in memory for the repeater as trusted.

I strongly prefer the second idea, but it is more effort. Since it's server side, it could eventually be expanded to be configured outside of NN and, if proven, possibly rolled into NN itself. I'm happy to try to implement it myself, but I don't want to step on toes, miss a major problem, or build something nobody wants. Thoughts?

**Comment by ThomasFreedman** [comment]
Yes, I think there is a use case for what you describe. I attend various meetups & festivals around the country and public msg boards for groups are very useful for attendees to post things for sale, talks being given etc. Having a wide open and readily available msg board to post such things provides a dynamic, on-the-fly public schedule of what & where things are going on in real time. Authentication for such a use case is actually not required and in fact is a barrier to its' use & success. 

Last year a public mesh network was implemented at Jackelope in AZ and Porcfest in NH for exactly this purpose. They would be much more useful if available through Sideband installed on smartphones.

**Comment by Swissbandit** [comment]
The message board is a real good solution.
However,a more private or closed group of allowed peers able to share among themselves would indeed be very nice to have.
There also is another solution offering a group chat,where the messages get delivered to the joined nn or sb peers directly,without being publicly on a nodepage. https://github.com/SebastianObi/LXMF-Tools/tree/main/lxmf_distribution_group

**Reply by faragher** [reply]
That actually looks like a mature and more feature-rich implementation than what I had in mind, apart from a self-registration of talk groups (which may exist, I didn't dig too deep). Honestly, I think both of my ideas have already been covered, maybe only with some self-register features which would be easier to add to an existing solution. 

This is amazing, thanks. I need to spend more time seeing what's currently available on the network.

**Comment by markqvist** [comment]
Closed groups natively within LXMF are not hard to make, per se.

**But** they are *really* hard to make *bandwidth efficient* while still preserving *privacy*. I think there can be an optimal solution, but it's something I need to lock myself in a room for a week to figure out, and I haven't had the oppertunity to do so yet ;)

One of these days, hopefully.

**Reply by TheTimC** [reply]
I'm loving having all you nerds on tap, I really appreciate all of your efforts! :-)

I'm really keen on getting a local and secure message board set up for my community. I've got a few rnodes running now on my heltec boards and I'm looking forward to when I can start rolloing these out in my area. The node to node comms is going to be great, but a group chat is something I think I will also need. 

Has anyone done a good guide yet on how to set up and host a message board / group chat?

Thanks.

**Reply by faragher** [reply]
Afraid not, so let's begin.

First, you need to determine your membership needs. If you're okay with anyone on the network having access, then any of the message board systems, or the distribution group, are great options. It may sound like this means they're public, and they can be, but you can also use IFAC to secure the network so only users with the passphrase can access the network. 

The security implications of a single password for the network is obvious.

If you want a private message board, either my board or the other one out there (the one where you log in, not the message wall style) can work for you. You'll need to create a user for the other variant.

Now, I want to be clear, I'm not actively supporting it at this time, but my message board was designed as a proof of concept. You can have public messages, viewable by anyone, and people who identify to the server can post in public and user-only fora. This means no accounts are required, and the only identifying information is the public identity of the user. 

It also has provisions for by-user authentication, so only listed identities can access specific sections. Admins can also delete posts and I can't recall if you ban users in config files or in the NomadNet interface,

In any case, any system running on NomadNet can also be restricted to authorized users using an .allowed file, but this has to be manually configured (although you could make a script.

Again, my board is not really supported or very user friendly, it's a tech demo of how I think a Reticulum-based message board should be handled. I haven't used the one that requires user profiles. The message wall one could work as a flat board where only people on the .allowed list are able to read it, but there's technically no provision to prevent people from *posting* to it, AFAIK. 

I'm a bit tired and feeling lazy right now, so I'm afraid I'm not going to track down all the links to these different implementations, but if any of these sound like the use case you want, and it sounds like the proper amount of work (I cannot reiterate enough how rough my implementation is) then I can expand on it tomorrow, hopefully.

**Reply by markqvist** [reply]
Link to the message board @faragher wrote: https://github.com/faragher/NomadLand

If anyone wants to add in links to other message board / forum solutions, please do so :)
